### ISSUE TITLE
`sinon.stub(obj)` should work even if obj happens to be a function

### ISSUE THREAD

**Issue by ethanresnick** [sinonjs/sinon#1967]
I'm not sure if this is a bug or a feature request; see below.

**Is your feature request related to a problem? Please describe.**
I have a class and I want to stub _all and only its **static** methods_. So, I want to treat the class's constructor function like a plain object, where I'm stubbing all the function-valued properties on it and its prototype (as those properties are the static methods).

More generally, I don't see why: 

```
function xyz() { return true; }
xyz.abc = function() { return false; }
xyz.__proto__.def = function() { return false; }

sinon.stub(xyz);
```

Shouldn't produce an object with `abc` and `def` replaced with stub methods. 

**Describe the solution you'd like**
The `sinon.stub(obj)` seems to say that it will stub all the function-valued properties on the object. But functions are objects too, and it doesn't stub their function-valued properties. So maybe this is actually a bug and not a feature request.

**Describe alternatives you've considered**
None; not sure I have enough context to consider alternatives. I'm just telling you what I expected, which didn't seem to work.

**Comment by ethanresnick** [comment]
Note: [this issue](https://github.com/sinonjs/sinon/issues/562) is vaguely related, as it shows the same error that I get when trying to do what I want. But the person in that issue is trying to stub the function  itself, rather than just trying to treat the function like any other bag of properties. Imo, for their case, the proper return value would be an empty object rather than the error (possibly with a warning logged that links to that issue).

**Comment by mantoni** [comment]
I agree that this should be supported. I'd also say it's a bug, but for a different reason: In latest Sinon 7.2.2 calling `sinon.stub(function () {})` throws

```
TypeError: Attempted to wrap undefined property undefined as function
    at wrapMethod (.../node_modules/sinon/lib/sinon/util/core/wrap-method.js:71:21)
```

This shows that this case is not properly handled. Sending a PR to fix in a second.

**Comment by mantoni** [comment]
This was fixed in Sinon `v7.2.3`.
